tensorboard/__main__.py

```diff
@@ -17,9 +17,10 @@
 from __future__ import division
 from __future__ import print_function
 
-import sys
+from tensorboard import main as tb_main
 
-from tensorboard.main import main
+run = tb_main.run
+main = tb_main.main
```

Comment:
Do we need the legacy "main" here?  I find it hard to imagine anyone would have been manually importing the symbol from `__main__.py` - in fact I'm not sure it's even possible.

Reply:
There's a small chance the reflection magic in `tb.app.run` might want to call the main() function in the `__main__` module. Couldn't hurt.

Reply:
We don't have to rely on chance when the code is open source :)  It doesn't look for a "main" function if you pass in your own "main" which tb_main.run does: https://github.com/tensorflow/tensorflow/blob/r1.4/tensorflow/python/platform/app.py#L44

IMO still better to leave it out.  Keeps users from having to wonder why it's there.

Reply:
In that case, maybe we can just delete this entire file in a separate PR? I still don't fully understand what it is, aside from the tf.run.main reflection.

Reply:
Ok, separate PR SGTM.

`__main__.py` is what gets executed by `python -m tensorboard` so I think we would still want it.  It's the "main" for a package like what the `if __name__ == '__main__'` snippet does for a module.  (As an aside, I think it is true that the if statement here is superfluous; we should just do `from tensorboard.main import run` and `run()` unconditionally.)